From dfecac944a9b0d17b3d8f7ffa43a52537f14b526 Mon Sep 17 00:00:00 2001
From: Meng Li <Meng.Li@windriver.com>
Date: Mon, 26 Dec 2022 21:25:02 +0800
Subject: [PATCH] configs: aptiv_cvc_sousa: enable CONFIG_FIT_SIGNATURE for
 secure boot verification

Signed-off-by: Meng Li <Meng.Li@windriver.com>

The ostree image requires to boot from script boot.scr, while secure
boot enabled, it failed:

    => source
    ## Executing script at 80000000
    Wrong image format for "source" command

From the Kconfig of u-boot[1]:

    WARNING: When relying on signed FIT images with a required signature
    check the legacy image format is disabled by default, so that unsigned
    images cannot be loaded. If a board needs the legacy image format
    support in this case, enable it using CONFIG_LEGACY_IMAGE_FORMAT.

Set CONFIG_LEGACY_IMAGE_FORMAT=y to boot from script

[1] https://source.codeaurora.org/external/autobsps32/u-boot/tree/Kconfig?h=release/bsp34.0-2020.04#n372

Upstream-Status: Inappropriate [WR Linux specific]

Signed-off-by: Hongxu Jia <hongxu.jia@eng.windriver.com>
---
 configs/aptiv_cvc_sousa_defconfig | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/configs/aptiv_cvc_sousa_defconfig b/configs/aptiv_cvc_sousa_defconfig
index 77ff58c9ce..6f6fb86d5f 100644
--- a/configs/aptiv_cvc_sousa_defconfig
+++ b/configs/aptiv_cvc_sousa_defconfig
@@ -79,3 +79,5 @@ CONFIG_MENU=y
 CONFIG_CMD_BOOTMENU=y
 CONFIG_XEN_SUPPORT=y
 CONFIG_NXP_HSE_SUPPORT=y
+CONFIG_FIT_SIGNATURE=y
+CONFIG_LEGACY_IMAGE_FORMAT=y
-- 
2.27.0

